docs/rules-cookbook.md

```diff
@@ -0,0 +1,519 @@
+            - '/public/'
+```
+
+### 1.4  Combining different matching methods
+
+Match paths that begin with either `/secret/` or `/topsecret/` EXCEPT the subdirectories `public`:
+
+```
+  matches:
+    all:
+      - paths:
+          begins_with:
+            - '/secret/'
+            - '/topsecret/'
+      - paths:
+          not: True
+          begins_with:
+            - '/secret/public/'
+            - '/topsecret/public/'
+```
+
+As above, but using `ends_with` for negative matching instead:
+```
+  matches:
+    all:
+      - paths:
+          begins_with:
+            - '/secret/'
+            - '/topsecret/'
+      - paths:
+          not: True
+          ends_with:
+            - '/public/'
+```
+### 1.5 `domain_default` matching
+
+The `matches` directive can only be present in a rule that does *NOT* have the `domain_default` property set, for obvious reasons. A rule that has `domain_default: True` will match any requests to that domain that are not matched by a rule for the same domain that does explicit matching.
+
+Complete rule example:
+
+```
+- description: www.domain.local - domain default
+  domains:
+    - www.mydomain.com
+  domain_default: True
+  actions:
+    backend:
+      origin: 'https://default-origin.localdomain'
+    header_southbound:
+      - set:
+        field: 'Host'
+        value: 'default-origin.localdomain'
+      - set:
+        field: 'X-Forwarded-Host'
+        value: 'www.mydomain.com'
+```
+
+## 2. Routing
+
+### 2.1 Routing to a single backend
+
+Routes all matched requests to the backend at `https://my-backend.mydomain.com`:
+
+```
+  actions:
+    backend:
+      origin: 'https://my-backend.mydomain.com'
+```
+
+### 2.2 Balancing over multiple backends
+
+Spreads the matched requests over two servers:
+
+```
+  actions:
+    backend:
+      servers:
+        - 'https://backend-1.mydomain.com'
+        - 'https://backend-2.mydomain.com'
+```
+
+### 2.3 Multiple backends with custom queue size and connection limits:
+
+```
+  actions:
+    backend:
+      servers:
+        - server: 'https://small-backend.mydomain.com'
+          max_connections: 32
+          max_queued_connections: 16
+        - server: 'https://large-backend.mydomain.com'
+          max_connections: 2048
+          max_queued_connections: 1024
+```
+
+## 3. Rewriting
+
+### 3.1 Rewriting paths
+
+Add a prefix to the request path:
+
+```
+  actions:
+    req_path:
+      - prefix:
+          add: /myprefix
+```
+
+Replace a specific part of a path regardless of case:
+
+```
+  actions:
+    req_path:
+      - replace:
+          from_exact: '/path/to/replace'
+          ignore_case: true
+          to: '/new/path'
+```
+
+Rewrite a path based on regular expression matching:
+
+```
+  actions:
+    req_path:
+      - replace:
+          from_regex: '/some/path/(.*)'
+          to_regsub: '/\1'
+```
+
+The above actually removes a the prefix '/some/path' from the path, which can also be achieved using the `prefix` structure:
```

Comment:
`s/a the/the/`

Reply:
Fixed in 4fcc40dfd6d8dfe14008aa4066cb3cfc1cc39af3